include/exec/linux/io_uring_context.hpp

```diff
@@ -0,0 +1,231 @@
+/*
+ * Copyright (C) 2023 Maikel Nadolski
+ * Copyright (c) 2023 NVIDIA Corporation
+ *
+ * Licensed under the Apache License Version 2.0 with LLVM Exceptions
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *   https://llvm.org/LICENSE.txt
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include "../../stdexec/execution.hpp"
+
+#include "../__detail/__atomic_intrusive_queue.hpp"
+#include "../__detail/__atomic_ref.hpp"
+
+#include "./safe_file_descriptor.hpp"
+#include "./memory_mapped_region.hpp"
+
+#if !__has_include(<linux/io_uring.h>)
+#error "io_uring.h not found. Your kernel is probably too old."
+#else
+#include <linux/io_uring.h>
+#endif
+
+#if !__has_include(<linux/version.h>)
+#error "linux/version.h not found. Do you use Linux?"
+#else
+#include <linux/version.h>
+#endif
+
+#if LINUX_VERSION_CODE < KERNEL_VERSION(5, 5, 0)
+#warning "Your kernel is too old to support io_uring with cancellation support."
+#else
+#define STDEXEC_HAS_IO_URING_ASYNC_CANCELLATION
+#endif
+
+#if LINUX_VERSION_CODE < KERNEL_VERSION(5, 6, 0)
+#include <sys/uio.h>
+#else
+#define STDEXEC_HAS_IORING_OP_READ
+#endif
+
+namespace exec {
+  struct schedule_after_t {
+    template <class _Scheduler, class _Duration>
+      requires stdexec::tag_invocable<schedule_after_t, const _Scheduler&, _Duration>
+    auto operator()(const _Scheduler& __scheduler, _Duration __duration) const
+      -> stdexec::tag_invoke_result_t<schedule_after_t, const _Scheduler&, _Duration> {
+      return tag_invoke(*this, __scheduler, __duration);
+    }
+  };
+
+  inline constexpr schedule_after_t schedule_after{};
```

Comment:
`schedule_after` is part of a `timed_scheduler` concept, which subsumes `scheduler`. We should factor this out into a `exec/timed_scheduler.hpp` header and define the concept there, since we'll have many more models of it and algorithms that will need to be constrained to accept only `timed_scheduler`s.

Reply:
Indeed, I wasn't sure where to put it and needed any CPO for now. I will do as you propose